openpype/tools/ayon_launcher/models/actions.py

```diff
@@ -359,7 +360,10 @@ def trigger_action(self, project_name, folder_id, task_id, identifier):
                     project_name, folder_id, task_id
                 )
                 force_not_open_workfile = per_action.get(identifier, False)
-                action.data["start_last_workfile"] = force_not_open_workfile
+                if force_not_open_workfile:
+                    action.data["start_last_workfile"] = False
+                else:
+                    action.data.pop("start_last_workfile", None)
```

Comment:
How come we are removing it altogether? If it's missing it will open the workfile?

Reply:
From launcher we want to control only to force to skip the workfile. There is also launch hook which fill the value based on  settings, but the settings are not applied if `"start_last_workfile"` already has any value.

Reply:
Just wondering - why is `start_last_workfile` there to begin with if we always want to either pop it here or set it here? :)

Reply:
Here, we want to have ability to set `start_last_workfile` to `False`, remember this as "the goal" with "Skip last workfile checkbox". The `pop` is here only to make sure it is not set in `action.data` because actions are cached for some time and launch of an app from one context could affect how it is launched from different context.